### PR TITLE
Copy response to clipboard #4300

### LINKED ISSUE #4300: Feature Request: copy response to clipboard
## Summary
After doing a request, it would be useful if there was a clipboard icon next to the response body that could copy the response body to the clipboard.

For large response bodies, it's quite tedious to try to select it all, cmd/ctrl+a tends to select the entire page. 

For our use cases, we often copy the response from 1 request, and then paste it into some other tool (e.g. GeoJSON editing), or use it as the request body to another request (e.g. a PUT request to update something.)

| Q                               | A
| ------------------------------- | -------
| Bug or feature request?         |  feature request
| Which Swagger/OpenAPI version?  | 2.0
| Which Swagger-UI version?       | 3.9.3
| How did you install Swagger-UI? | We use the static files, and just modify the javascript in `index.html` to change the settings as needed.
| Which browser & version?        | Chrome...latest
| Which operating system?         | Mac OS

### Demonstration API definition
n/a

### Configuration (browser query string, constructor, config.yaml)
n/a

### Expected Behavior
There is a clipboard icon that you can click to copy the response received into the operating system's clipboard.

### Current Behavior
There is no such icon

### Possible Solution
* This could be configurable so that it's not "always on"
* Could fix ctrl/cmd+a so that it selects just the response body if the cursor is inside the response body

### Context
This feature would make it much easier to interact with the responses from Swagger UI in various ways.

### PR DESCRIPTION
**Pull request by aldrinabastillas**
### Description
Adds a button to copy response to clipboard next to the download button.
Cherry-picked from open PR 4312, updated styling and of responses the button should be available for

### Motivation and Context
Fixes #4300
Reduces scrolling when having to Ctrl+A + Ctrl+C for large payloads

### How Has This Been Tested?
Manually tested in ui with json, xml, html, and text responses.

### Screenshots (if appropriate):
image

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [x] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [x] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [x] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/core/components/response-body.jsx [hunk 1]
```diff
@@ -99,19 +99,19 @@ export default class ResponseBody extends React.PureComponent {
         body = "can't parse JSON.  Raw result:\n\n" + content
       }
 
-      bodyEl = <HighlightCode downloadable fileName={`${downloadName}.json`} value={ body } />
+      bodyEl = <HighlightCode downloadable fileName={`${downloadName}.json`} value={ body } canCopy />
```
Comment (heldersepu) [inline src/core/components/response-body.jsx]: Where is canCopy declared? Is it needed?

**Comment by aldrinabastillas** [comment]
Sorry for the late response! Comment got filtered in my inbox; let me know if I have other changes to make or need to clean up the branch at all.

**Comment by heldersepu** [comment]
Some unit tests would be nice

**Comment by heldersepu** [comment]
@shockey  on the previous PR you mentioned: 
> I've run this up the flagpole for a UX review
But that was a year ago, any chance this can make it?

**Comment by vgiurgiu** [comment]
Hi @aldrinabastillas  - we have a set of Swagger APIs and customers are asking about a "copy to clipboard" functionality. 

My Google searches led me to this pull request and I can't tell by the comments or commits how close it is to merging. Or is this functionality already available? 

Thanks!
Victor

**Comment by heldersepu** [comment]
@vgiurgiu  the fact that the last comment was `Apr 22, 2019` that tell us this does not seem to be a priority for the team

**Comment by aabastillas-rca** [comment]
Hi all, I simply just merged master back in. Not sure who else on the team needs to approve before it's okay to close out the PR.

**Comment by tim-lai** [comment]
@aldrinabastillas @aabastillas-rca Merged! Thanks for the contribution and patience to get this merged!

@heldersepu Thanks for the review, it helped get this PR merged!

**Pull request merged**